### PR TITLE
refactor: dry some timings calculation

### PR DESCRIPTION
**Pull request by steve-chavez**

### CODE REVIEW

File: src/PostgREST/App.hs [hunk 1]
```diff
@@ -173,58 +170,28 @@ handleRequest :: AuthResult -> AppConfig -> AppState.AppState -> Bool -> Bool ->
...
-      (respTime', pgrst) <- withTiming $ liftEither $ Response.invokeResponse cPlan invMethod (Plan.crProc cPlan) apiReq resultSet
-      return $ pgrstResponse (ServerTiming jwtTime parseTime planTime' txTime' respTime') pgrst
-
-    ActSchema tSchema (ActSchemaRead headersOnly) -> do
-      (planTime', iPlan) <- withTiming $ liftEither $ Plan.inspectPlan apiReq
-      (txTime', oaiResult) <- withTiming $ runQuery roleIsoLvl mempty (Plan.ipTxmode iPlan) $ Query.openApiQuery sCache pgVer conf tSchema
-      (respTime', pgrst) <- withTiming $ liftEither $ Response.openApiResponse (T.decodeUtf8 prettyVersion, docsVersion) headersOnly oaiResult conf sCache iSchema iNegotiatedByProfile
-      return $ pgrstResponse (ServerTiming jwtTime parseTime planTime' txTime' respTime') pgrst
-
-    ActRelation identifier ActRelInfo -> do
```
Comment (steve-chavez) [inline src/PostgREST/App.hs:218]: Reduced some timings calculation here. This is just some progress. The final goal is to calculate them once on each stage (plan, query, response).

File: src/PostgREST/ApiRequest.hs [hunk 2]
```diff
@@ -92,23 +90,17 @@ data Resource
...
-  = ActInvoke InvokeMethod
-  | ActRoutInfo
-
-data ActionSchema
-  = ActSchemaRead Bool
-  | ActSchemaInfo
+data DbAction
+  = ActRelationRead {dbActQi :: QualifiedIdentifier, actHeadersOnly :: Bool}
+  | ActRelationMut  {dbActQi :: QualifiedIdentifier, actMutation :: Mutation}
+  | ActRoutine      {dbActQi :: QualifiedIdentifier, actInvMethod :: InvokeMethod}
```
Comment (steve-chavez) [inline src/PostgREST/ApiRequest.hs:96]: codecov fails because of these unused records.

File: src/PostgREST/ApiRequest.hs [hunk 3]
```diff
@@ -92,23 +90,17 @@ data Resource
...
 
 data Action
-  = ActRelation QualifiedIdentifier ActionRelation
-  | ActRoutine QualifiedIdentifier ActionRoutine
-  | ActSchema Schema ActionSchema
+  = ActDb           DbAction
+  | ActSchemaRead   Schema Bool
+  | ActRelationInfo QualifiedIdentifier
+  | ActRoutineInfo  QualifiedIdentifier
+  | ActSchemaInfo   Schema
```
Comment (laurenceisla) [inline src/PostgREST/ApiRequest.hs:103]:
```suggestion
  | ActSchemaInfo
```

I think the schema is not needed anymore here.

Comment (steve-chavez) [reply src/PostgREST/ApiRequest.hs:103]: Will address that in a new PR

**Comment by steve-chavez** [comment]
Hm, everything is failing on main https://github.com/PostgREST/postgrest/actions/runs/8459962842.

Seems it's about the removal of `NamedFieldPuns` on another PR.\

Will fix.

**Pull request merged**